src/org/javarosa/core/model/FormDef.java

```diff
@@ -353,6 +353,10 @@ public TreeReference getChildInstanceRef(List<IFormElement> elements, List<Integ
         return ref;
     }
 
+    public TreeElement getFirstDescendantWithName(String name) {
```

Comment:
If this is only used for testing, it would be better to have it in another separate class under the test sources root. It will make our production classes thinner and more manageable and will clearly set the right context (testing vs. production) for all future readers.

Moving this to another separate class under the test sources root would also contribute towards our goal for a test harness by adding more testing use cases in a centralized place, which will make easier to make sense of the testing needs in JR.